src/scipp/io/hdf5.py

```diff
@@ -215,7 +215,7 @@ class DataArrayIO:
     @staticmethod
```

Comment:
I think it all looks good but we should wait until @SimonHeybrock has had a look before merging.